File: 2022/02/a.py
f = open("input.txt", "r")

ROCK = 1
PAPER = 2
SCISSORS = 3
WIN = 6
DRAW = 3
LOSS = 0
OUTCOMES_A = {
        "A X": ROCK + DRAW,
        "A Y": PAPER + WIN,
        "A Z": SCISSORS + LOSS,
        "B X": ROCK + LOSS,
        "B Y": PAPER + DRAW,
        "B Z": SCISSORS + WIN,
        "C X": ROCK + WIN,
        "C Y": PAPER + LOSS,
        "C Z": SCISSORS + DRAW,
}


def solution_one():
    score = 0
    for line in f:
        score += OUTCOMES_A[line.strip()]
    print(score)


OUTCOMES_B = {
        "A X": SCISSORS + LOSS,
        "A Y": ROCK + DRAW,
        "A Z": PAPER + WIN,
        "B X": ROCK + LOSS,
        "B Y": PAPER + DRAW,
        "B Z": SCISSORS + WIN,
        "C X": PAPER + LOSS,
        "C Y": SCISSORS + DRAW,
        "C Z": ROCK + WIN,
}


def solution_two():
    score = 0
    for line in f:
        score += OUTCOMES_B[line.strip()]
    print(score)


solution_two()

File: 2022/02/a.cpp
#include <iostream>
#include <sstream>

using namespace std;

void solution_one() {
	string line;
	int score = 0;
	while(getline(cin, line)) {
		istringstream iss(line);
		char opp, me;
		iss >> opp >> me;
		if(me == 'X') {
			score += 1;
			if(opp == 'A') {
				score += 3;
			} else if(opp == 'B') {
				score += 0;
			}
			else if(opp == 'C') {
				score += 6;
			}
		}
		else if(me == 'Y') { 
			score += 2;
			if(opp == 'A') {
				score += 6;
			} else if(opp == 'B') {
				score += 3;
			}
			else if(opp == 'C') {
				score += 0;
			}
		}
		else if(me == 'Z') {
			score += 3;
			if(opp == 'A') {
				score += 0;
			} else if(opp == 'B') {
				score += 6;
			}
			else if(opp == 'C') {
				score += 3;
			}
		}
	}
	cout << score << endl;
}

void solution_two() {
	string line;
	int score = 0;
	while(getline(cin, line)) {
		istringstream iss(line);
		char me, opp;
		iss >> opp >> me;
		if(me == 'X') {
			score += 0;
			if(opp == 'A') {
				score += 3;
			} else if(opp == 'B') {
				score += 1;
			}
			else if(opp == 'C') {
				score += 2;
			}
		}
		else if(me == 'Y') { 
			score += 3;
			if(opp == 'A') {
				score += 1;
			} else if(opp == 'B') {
				score += 2;
			}
			else if(opp == 'C') {
				score += 3;
			}
		}
		else if(me == 'Z') {
			score += 6;
			if(opp == 'A') {
				score += 2;
			} else if(opp == 'B') {
				score += 3;
			}
			else if(opp == 'C') {
				score += 1;
			}
		}
	}
	cout << score << endl;
}

int main() {
	solution_two();
}
